src/main/java/dev/hephaestus/atmosfera/client/sound/modifiers/implementations/BossBarCondition.java

```diff
@@ -0,0 +1,55 @@
+package dev.hephaestus.atmosfera.client.sound.modifiers.implementations;
+
+import com.google.gson.JsonObject;
+import dev.hephaestus.atmosfera.client.sound.modifiers.AtmosphericSoundModifier;
+import dev.hephaestus.atmosfera.world.context.EnvironmentContext;
+import net.minecraft.client.gui.hud.ClientBossBar;
+import net.minecraft.entity.boss.BossBar;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.world.World;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+
+public record BossBarCondition(String text, boolean use_regex) implements AtmosphericSoundModifier, AtmosphericSoundModifier.Factory {
+
+    @Override
+    public float getModifier(EnvironmentContext context) {
+        if(!(text instanceof String)) {
+            return 0;
+        }
```

Comment:
A null check would be more appropriate here, but even so, `text` should not be null at this point. That's the kind of thing you'd want to check in the deserialization method, `create(JsonObject)`.